source/Halibut.Tests/AsyncServicesFixture.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Tests.Support.TestCases;
+using Halibut.Tests.TestServices.Async;
+using Halibut.Tests.Util;
+using Halibut.TestUtils.Contracts;
+using Halibut.Util;
+using NUnit.Framework;
+
+namespace Halibut.Tests
+{
+    public class AsyncServicesFixture : BaseTest
+    {
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testAsyncServicesAsWell: true, testNetworkConditions: false)]
+        public async Task AsyncServiceWithParams_CanBeRegisteredAndResolved(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            var clientAndServiceBuilder = clientAndServiceTestCase.CreateTestCaseBuilder()
+                .AsLatestClientAndLatestServiceBuilder();
+
+            if (clientAndServiceTestCase.ServiceAsyncHalibutFeature.IsEnabled())
+            {
+                clientAndServiceBuilder = clientAndServiceBuilder
+                    .WithAsyncService<IEchoService, IAsyncEchoService>(() => new AsyncEchoService());
+            }
+            else
+            {
+                clientAndServiceBuilder = clientAndServiceBuilder
+                    .WithService<IEchoService>(() => new EchoService());
+            }
+
+            var clientAndService = await clientAndServiceBuilder.Build(CancellationToken);
+
+            var value = Some.RandomAsciiStringOfLength(8);
+            var echoServiceClient = clientAndService.CreateClient<IEchoService, IAsyncClientEchoService>();
+            var result = await echoServiceClient.SayHelloAsync(value);
+
+            if (clientAndServiceTestCase.ServiceAsyncHalibutFeature.IsEnabled())
+            {
+                result.Should().Be($"{value}Async...");
+            }
+            else
+            {
+                result.Should().Be($"{value}...");
+            }
+        }
+
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testAsyncServicesAsWell: true, testNetworkConditions: false)]
+        public async Task AsyncServiceWithNoParams_CanBeRegisteredAndResolved(ClientAndServiceTestCase clientAndServiceTestCase)
```

Comment:
Should be this be calling a service that has a method that takes no arguments?

Reply:
Ah it's poorly named - it's for testing that we can invoke an async method with `Task` for return type (rather than `Task<T>`)

I can rename these tests, but then that makes me realise I should probably have tests for the _actual_ params/no params cases. Should I add those tests in a separate PR after this one, to prevent this one dragging on longer?

Reply:
Best do it now in case it digs out some bugs